handlers/api-handler.ts

```diff
@@ -44,13 +44,14 @@ export interface ApiHandlerDefinition<B = never, Q = never, R = never>
 	};
 }
 
-export type ApiHandlerAuthorizer<A> = (
-	event: APIGatewayProxyEventV2,
+export type ApiHandlerAuthorizer<B, Q, P extends ReadonlyArray<string>, A, R> = (
```

Comment:
Do we have any more constraints these generics should be following? i.e. are these things at least 'object' types?

B extends {} Q extends {}, etc. I don't have much context here but it seems like we should include any constraints we do know of

Reply:
These generics are controlled by what is passed into the `ApiHandler` options using the `schema`. If there is no schema for "query" passed, then we shouldn't expect a type for it. RIght now, it's shown as "unknown" but I could see a case where they should be "never". I'd have to do some digging though to see what the behavior is for these attributes if they don't have an associated validator.

Regardless though, this is out of the scope of the PR